File: resources/scop_1.5.1/java/examples/XMLClient.java
// XMLClient.java - DMI - 25-12-02

/* Usage: java XMLClient [<n> <k>]    (default values n = 4, k = 2) */

import java.io.*;

public class XMLClient
{
	public static void main(String[] args)
	{
		SCOPXML scop;
		int n, k;
		Vertex v, w;

		if(args.length != 2)
		{
			n = 4;
			k = 2;
		}
		else
		{
			n = Integer.parseInt(args[0]);
			k = Integer.parseInt(args[1]);
		}

		scop = new SCOPXML("localhost", "XMLClient");	
		v = Vertex.pack(Vertex.pack(n), Vertex.pack(k));
		w = scop.rpc("XMLServer", v);
		System.out.println(n + " choose " + k + " equals " + w.extract_int()
				+ ".");
		scop.close();
	}
}

File: resources/scop_1.5.1/python/examples/xml_client.py
#!/usr/bin/python
# xml_client.py - DMI - 11-11-03

# Usage: xml_client [<n> <k>]    (default values n = 4, k = 2)

import scop, sys

if len(sys.argv) != 3:
	n, k = 4, 2
else:
	n, k = int(sys.argv[1]), int(sys.argv[2])
sock = scop.scop_open("localhost", "xml_client")
v = [n, k]
w = scop.scop_rpc(sock, "xml_server", v)
print str(n) + " choose " + str(k) + " equals " + str(w) + "."
scop.scop_close(sock)
